### DISCUSSION TITLE
How to trigger serverless function on deploy?

### DISCUSSION THREAD

**Discussion by remjx** [vercel/vercel#5029] [Help] [answered]
I am creating a database migration function using [umzug](https://www.npmjs.com/package/umzug). Instead of triggering the function manually via HTTP, I want it to run after (or during) every deploy. The function checks the database to see if there any pending migrations, and if so, executes them.

Is there any way to accomplish this as part of Vercel [Deployments](https://vercel.com/docs/platform/deployments)?

**Comment by styfle** [comment, marked as answer]
I would suggest running the database migration in the [build step](https://vercel.com/docs/build-step) so it will execute for every deploy.

**Reply by rvirani1** [reply]
I would just make sure it occurs after the main build step in that case. You don't want to run migrations and then find that your build failed. 

Assuming you have a script in your `package.json` called `migrate` to run migrations.

```
next build && npm run migrate
```
